config_gen.py

```diff
@@ -202,6 +202,25 @@ def run(cmd, *args, **kwargs):
         print("$ " + " ".join(cmd))
         subprocess.call(cmd, *args, **kwargs)
 
+    # helper function to set CC and CXX variables correctly on OSX with qmake
+    def patch_qmake_makefile(makefile_path):
+        compilers = {
+            "CC": "clang",
+            "CXX": "clang++",
+            "LINK": "clang++"
+        }
+        patched_makefile_lines = []
+        with open(makefile_path, "r") as f:
```

Comment:
I used a combination of `match` and `sub` because so the spaces before `=` are untouched. In the patched Makefile the equal signs are stilled aligned. The `match` is only used for a better style of the patched result and the code block could be replaced by a single `sub` call as you commented before. Or is it possible to use `sub` without changing the style of the original Makefile?